ably/options.go

```diff
@@ -699,6 +702,15 @@ func WithDial(dial func(protocol string, u *url.URL, timeout time.Duration) (pro
 	}
 }
 
+// WithBaseContext establishes a context to use for operations initiated by the
+// the library, such as establishing realtime connections and requesting
+// implicit authorizations.
```

Comment:
It's bad practice to store a context as a struct field, is passing a context directly to function calls not sufficient?

Reply:
The library makes requests on its own, from the goroutine that manages the connection, for reauth, reconnection, etc. It can't take the context from the caller's stack because the caller is the library itself.

Now that I think of it, I don't think having a custom base context is useful at all; users can provide `Dial` and `http.Client` options that ultimately are passed and can wrap the context, so we might as well just pass `context.Background()` to them. WDYT?

Reply:
Removed the option at 3b04db76d797942ccfb586ab52e1b3fdcc9b37bb.

Reply:
Yep makes sense 👍